controllers/comments.js

```diff
@@ -17,7 +17,14 @@ exports.getCommentsByArticleId = (req, res, next) => {
 };
 
 exports.postComment = ({ params, body }, res, next) => {
-  insertComment(params.article_id, body)
+  const promises = [
+    checkExists("articles", "article_id", params.article_id),
+    checkExists("users", "username", body.username),
+  ];
+  Promise.all(promises)
+    .then(() => {
+      return insertComment(params.article_id, body);
```

Comment:
You may also find Duncan that you don't explicitly need to check for the existence of either a user or the article due to the FOREIGN KEY restrictions on the comments table

Reply:
I did it in part this way so that I could return a 404 as they associated data would not be found. Should I be returning a 400 bad request if they try to post with invalid FK's?